wsl-exec: Ensure a minimum set of PATH

jandubois opened the pull request:
If we override PATH when spawned via Windows (via setting WSLENV), the default set of entries do not get picked up because it makes no sense on the Windows side.  This means that basic system utilities end up missing; this causes issues like runc not being able to find iptables.

When wsl-exec is run, ensure that a hard-coded list is found.

mook-as commented on pkg/rancher-desktop/assets/scripts/wsl-exec:
```diff
@@ -9,6 +9,14 @@
 set -o errexit -o nounset
 
+# We may have WSLENV set to override PATH; however, that causes the default entries to be missing.
+IFS=':' read -ra dirs <<< /usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
```
The shebang says `/bin/sh`; this seems to contain bashisms?

jandubois replied:
Ah, yes, good catch

jandubois replied:
New variant tested like this:

```
/mnt/c/Users/SUSE # cat addpath.sh
str="/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:"
while [ -n "$str" ]; do
  dir=${str%%:*}
  if ! [[ ":${PATH}:" =~ :${dir}: ]]; then
    export PATH="${PATH%:}:${dir}"
  fi
  str=${str#*:}
done
/mnt/c/Users/SUSE # PATH=/bin
/mnt/c/Users/SUSE # echo $PATH
/bin
/mnt/c/Users/SUSE # . addpath.sh
/mnt/c/Users/SUSE # echo $PATH
/bin:/usr/local/sbin:/usr/local/bin:/sbin:/usr/sbin:/usr/bin
```

jandubois committed:
wsl-exec: Ensure a minimum set of PATH

If we override PATH when spawned via Windows (via setting WSLENV), the
default set of entries do not get picked up because it makes no sense on
the Windows side.  This means that basic system utilities end up missing;
this causes issues like runc not being able to find iptables.

When wsl-exec is run, ensure that a hard-coded list is found.

Authored by Mark Yen <mark.yen@suse.com>

Signed-off-by: Jan Dubois <jan.dubois@suse.com>
